monitoring/docs/monitoring-and-telemetry.adoc

```diff
@@ -0,0 +1,206 @@
+The dashboard instances collect the following telemetry data that are sent
+directly to Sentry hub:
+
+==== Generated deposit data
+
+Each time a new deposit address is generated, the dashboard collects the
+following data:
+
+* BTC address of the deposit
+* JSON recovery file content
+* Second track deposit verification result
+
+Worth noting that those data are send to the telemetry regardless of whether
+the user actually funded and revealed the deposit. Those data are collected
+in order to help users with potential deposit reveals that have not occurred
+automatically and deposit recovery in case of wallet misbehavior. Last but not
+least, all deposit addresses are checked against the second track deposit
+verifier to make sure the address generation logic used by the dashboard is
+proper and will not lead to fund loss.
+
+[NOTE]
+====
+The second track deposit verifier is actually a Google Cloud Function named
+https://console.cloud.google.com/functions/details/us-central1/verify-deposit-address?env=gen1&authuser=1&project=keep-prd-210b[`verify-deposit-address`]
+living in the `keep-prd` GCP project. That function accepts some deposit data
+and return a deposit BTC address computed using their own algorithm,
+independent from the one used by the dashboard. See an https://us-central1-keep-prd-210b.cloudfunctions.net/verify-deposit-address/json/mainnet/latest/1fb2d377340f4b776ed2516b2293bd65fd205858/abb75eb77b3eb032/5efe583da177c3a421578a9a5ee4ad4837991474/390e4165[example call].
+====
+
+==== Unhandled errors
+
+Apart from deposit data, all unhandled errors thrown by the dashboard are also
+captured by the telemetry system. This allows to identify bugs encountered by
+users in an automatic way.
+
+=== TBTCv2 minters and guardians
+
+The minters and guardians instances collect the following telemetry data that
+are sent directly to Sentry hub:
+
+==== Handled processing and validation errors
+
+During their work, the minters and guardians may encounter several errors
+related to the processing logic and deposit validation. Most of them are
+explicitly sent to the telemetry as errors or warnings. Examples of such
+errors are:
+
+* Revert of an Ethereum transaction created by the bot
+* Failed validation of a revealed deposit
+
+==== Unhandled errors
+
+Apart from handled errors collected explicitly, all unhandled errors thrown by
+the minters and guardians are also captured by the telemetry system.
+
+=== TBTCv2 monitoring
+
+The monitoring component inspects the tBTC v2 system on-chain contracts and
+produce different kind of system events that are sent to Sentry and Discord
+based on their type. A general rule of thumb is that notifications
+(*i.e. informational sytem events*) are sent directly to Discord `pass:[*]-notifications`
+channels while alerts requiring an action (*i.e. warning/critical system events*)
+are propagated to the Sentry hub that decides about next steps. Specific system
+events produced by the monitoring component are:
```

Comment:
```suggestion
events produced by the monitoring component are:

* deposit revealed,
* wallet registered,
* large deposit revealed,
* optimistic minting canceled,
* optimistic minting requested too early,
* optimistic minting requested for undetermined Bitcoin transaction.
```

It is not a huge problem but I think this version is easier to understand by someone who reads about the monitoring for the first time.

Reply:
See e769513f